fetch/api/headers/header-values-normalize.html

```diff
@@ -22,43 +22,49 @@
     strip = true
   }
 
-  let val = String.fromCharCode(i),
-      expectedVal = strip ? "" : val
-  for(let y = 0; y < 3; y++) {
-    let useVal = val,
-        useExpectedVal = expectedVal
-    if(y === 0) {
-      useVal = "x" + val
-      useExpectedVal = "x" + expectedVal
-    } else if(y === 2) {
-      useVal = val + "x"
-      useExpectedVal = expectedVal + "x"
-    }
+  let url = "../resources/inspect-headers.py?headers=val1|val2|val3",
+      val = String.fromCharCode(i),
+      expectedVal = strip ? "" : val,
+      val1 = val,
+      expectedVal1 = expectedVal,
+      val2 = "x" + val,
+      expectedVal2 = "x" + expectedVal,
+      val3 = val + "x",
+      expectedVal3 = expectedVal + "x"
 
-    async_test((t) => {
-      let xhr = new XMLHttpRequest()
-      xhr.open("POST", "../resources/inspect-headers.py?headers=value-test")
-      if(fail) {
-        assert_throws("SyntaxError", () => xhr.setRequestHeader("value-test", useVal))
+  async_test((t) => {
+    let xhr = new XMLHttpRequest()
+    xhr.open("POST", url)
+    if(fail) {
+        assert_throws("SyntaxError", () => xhr.setRequestHeader("val1", val1))
+        assert_throws("SyntaxError", () => xhr.setRequestHeader("val2", val2))
+        assert_throws("SyntaxError", () => xhr.setRequestHeader("val3", val3))
         t.done()
-      } else {
-        xhr.setRequestHeader("value-test", useVal)
-        xhr.onload = t.step_func_done(() => {
-          assert_equals(xhr.getResponseHeader("x-request-value-test"), useExpectedVal)
-        })
-        xhr.send()
-      }
-    }, "XMLHttpRequest with value " + encodeURI(useVal))
+    } else {
+      xhr.setRequestHeader("val1", val1)
+      xhr.setRequestHeader("val2", val2)
+      xhr.setRequestHeader("val3", val3)
+      xhr.onload = t.step_func_done(() => {
+        assert_equals(xhr.getResponseHeader("x-request-val1"), expectedVal1)
+        assert_equals(xhr.getResponseHeader("x-request-val2"), expectedVal2)
+        assert_equals(xhr.getResponseHeader("x-request-val3"), expectedVal3)
+      })
+      xhr.send()
+    }
+  }, "XMLHttpRequest with value " + encodeURI(val))
 
-    promise_test((t) => {
-      if(fail) {
-        return promise_rejects(t, new TypeError(), fetch("about:blank", { headers: {"value-test": useVal} }))
-      } else {
-        return fetch("../resources/inspect-headers.py?headers=value-test", { headers: {"value-test": useVal} }).then((res) => {
-          assert_equals(res.headers.get("x-request-value-test"), useExpectedVal)
-        })
-      }
-    }, "fetch() with value " + encodeURI(useVal))
-  }
+  promise_test((t) => {
+    if(fail) {
+      promise_rejects(t, new TypeError(), fetch("about:blank", { headers: {"val1": val1} }))
+      promise_rejects(t, new TypeError(), fetch("about:blank", { headers: {"val2": val2} }))
+      return promise_rejects(t, new TypeError(), fetch("about:blank", { headers: {"val3": val3} }))
```

Comment:
Does that work? I cannot really find any documentation about how this is supposed to work. @jgraham?